contracts/reflect/src/contract.rs

```diff
@@ -259,4 +271,26 @@ mod tests {
             _ => panic!("Must return unauthorized error"),
         }
     }
+
+    #[test]
+    fn dispatch_custom_query() {
+        // stub gives us defaults. Consume it and override...
```

Comment:
I just tried

```rust
        let deps = Extern {
            querier: CustomQuerier {},
            ..mock_dependencies(20, &[])
        };
```

which does not work because it is not only "overriding", but creating a different Extern type with different generic types.

Now I wonder how valuable the usage of `mock_dependencies` is, when you don't use the balance argument at all.

Reply:
I often do use the balance argument, but yeah the constructors could definitely use some work. They were there to help, but seem to limit too much when wanting to extend. Also look at mock interface with the same hoops , or even more